mailpoet/assets/js/src/newsletters/send/date-text.tsx

```diff
@@ -155,16 +155,12 @@ class DateText extends Component<DateTextProps> {
       .replace(/\]/g, '');
   };
 
-  getDate = (date: string) => {
-    const formatting = {
-      parseFormat: this.props.storageFormat,
-    } as DateOptions;
-    return MailPoet.Date.toDate(date, formatting);
-  };
+  getDate = (date: string) => Moment(date).toDate();
 
-  getStorageDate = (date: Date) => {
+  getAsStringInFormat = (date: Date) => {
     const formatting = {
       format: this.props.storageFormat,
+      offset: 0, // set 0 to prevent applying offset. At this point we want to return value exactly as picked in UI.
```

Comment:
Checking the implementation:

<img width="539" alt="Screenshot 2024-08-23 at 14 17 59" src="https://github.com/user-attachments/assets/a3bf6b6e-3d75-4f11-ad5e-bc4ba45f2fe7">



And the [Moment.js docs](https://momentjscom.readthedocs.io/en/latest/moment/03-manipulating/09-utc-offset/):

<img width="846" alt="Screenshot 2024-08-23 at 14 18 45" src="https://github.com/user-attachments/assets/6d92e964-9d0a-4c0c-a50a-29f72200ddeb">

It seems to me that this will preserve the point in time (Universal Time), but the display time will change:

> Passing false (the default) will keep the same instant in Universal Time, but the local time will change.

However, the comment in the code seems to suggest that it's not what you want:

> // set 0 to prevent applying offset. At this point we want to return value exactly as picked in UI.

So, that's a bit confusing to me. Maybe it's just me interpreting the comment wrong? What you're actually doing here is converting the time to UTC and then formatting that. Is that the desired behavior?

Reply:
Ahh, maybe in combination with Moment.utc(date) ([link](https://momentjscom.readthedocs.io/en/latest/moment/01-parsing/13-utc/)), this has no effect and only prevents some potential automatic conversion?

Reply:
> Ahh, maybe in combination with Moment.utc(date) ([link](https://momentjscom.readthedocs.io/en/latest/moment/01-parsing/13-utc/)), this has no effect and only prevents some potential automatic conversion?

Yes. I was thinking about it like that, but I will rather double check.